src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssCopy.java

```diff
@@ -9,5 +9,124 @@
+import zowe.client.sdk.rest.ZoweRequestFactory;
+import zowe.client.sdk.rest.type.ZoweRequestType;
+import zowe.client.sdk.utility.RestUtils;
+import zowe.client.sdk.utility.ValidateUtils;
+import zowe.client.sdk.zosfiles.ZosFilesConstants;
+import zowe.client.sdk.zosfiles.uss.input.CopyParams;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+
+/**
+ * Provides Unix System Services (USS) copy functionality
+ *
+ * @author James Kostrewski
+ * @version 2.0
+ */
 public class UssCopy {
+    private static final Logger LOG = LoggerFactory.getLogger(UssCopy.class);
+    private final ZosConnection connection;
+    private ZoweRequest request;
+
+    /**
+     * UssCopy Constructor
+     *
+     * <a href="https://www.ibm.com/docs/en/zos/2.4.0?topic=interface-zos-unix-file-utilities">z/OSMF REST API</a>
+     * @param connection connection information, see ZosConnection object
+     * @author James Kostrewski
+     */
+    public UssCopy(ZosConnection connection) {
+        ValidateUtils.checkConnection(connection);
+        this.connection = connection;
+    }
+
+    /**
+     * Alternative UssCopy constructor with ZoweRequest object. This is mainly used for internal code
+     * unit testing with mockito, and it is not recommended to be used by the larger community.
+     *
+     * @param connection connection information, see ZosConnection object
+     * @param request    any compatible ZoweRequest Interface object
+     * @author James Kostrewski
+     */
+    public UssCopy(ZosConnection connection, ZoweRequest request) {
+        ValidateUtils.checkConnection(connection);
+        this.connection = connection;
+        this.request = request;
+    }
+
+    private static String buildBody(CopyParams params) {
+        final Map<String, Object> jsonMap = new HashMap<>();
+
+        jsonMap.put("request", "copy");
+        if (params.getFrom().isPresent()) {
+            jsonMap.put("from", params.getFrom().get());
+        }
+        if (params.getFrom_dataset().isPresent()) {
+            jsonMap.put("from-dataset", params.getFrom_dataset().get());
+        }
+        if (params.isOverwrite() == false) {
+            jsonMap.put("overwrite", "false");
+        }
+        if (params.isRecursive()) {
+            jsonMap.put("recursive", "true");
+        }
+
+        final JSONObject jsonRequestBody = new JSONObject(jsonMap);
+        LOG.debug(String.valueOf(jsonRequestBody));
+        return jsonRequestBody.toString();
+    }
+
+    /**
+     * Copy a USS file or directory to another location
+     *
+     * @param destinationPath the destination path of the file or directory to copy
+     * @param sourcePath      the source path of the file or directory to copy
+     * @return ZoweRequest object with the response from the server
+     * @throws Exception if the request fails
+     */
+    public Response copy(String destinationPath, String sourcePath) throws Exception {
+        CopyParams.Builder builder = new CopyParams.Builder();
```

Comment:
use the .build() pattern here see the .build code in other places in the project.